### PR TITLE
Extract sample dataset metadata in Cypress

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
READY

### What does this PR accomplish?
1. It dynamically extracts sample dataset metadata (`table` ids, and accompanied `table_field` ids) during snapshot creating and seeding phase
2. Stores them in a json file at the beginning of every Cypress run (thus, the file is ignored by Git)
3. Makes it available for use in every Cypress test (through `frontend/test/__support__/cypress_sample_dataset.js`) as `SAMPLE_DATASET` const

### Example of usage
```javascript
// before
import { foo, bar, withSampleDataset } from "__support__/cypress";

describe("some > description > block", () => {
  it("tests something", ()=> {
    withSampleDataset(({ ORDERS, ORDERS_ID })) => {
      someFunctionThatUses(ORDERS);
      const someObject = {
        baz: ORDERS_ID,
      },
    });
  });

  it("tests something else", ()=> {
    withSampleDataset(({ ORDERS, ORDERS_ID, PRODUCTS })) => {
      someFunctionThatUses(PRODUCTS);
      const someObject = {
        baz: ORDERS.CREATED_AT,
        buz: ORDERS_ID,
      },
    });
  });
});
```

```javascript
// now
import { foo, bar } from "__support__/cypress";
import { SAMPLE_DATASET } from "__support__/cypress_sample_dataset";

const { ORDERS, ORDERS_ID, PRODUCTS } = SAMPLE_DATASET;

describe("some > description > block", () => {
  it("tests something", ()=> {
    someFunctionThatUses(ORDERS);
    const someObject = {
      baz: ORDERS_ID,
    },
  });

  it("tests something else", ()=> {
    someFunctionThatUses(PRODUCTS);
    const someObject = {
      baz: ORDERS.CREATED_AT,
      buz: ORDERS_ID,
    },
  });
});
```

### CODE REVIEW

**Comment by nemanjaglumac** [comment]
CI errored with
```
/home/circleci/********/********/frontend/test/__support__/cypress.js
  4:43  error  Unable to resolve path to module './cypress-sample-dataset.json'  import/no-unresolved
```

I tried to use an absolute path (aliased one) - that one failed as well. 

My best guess is that it fails because linter is running first - before Cypress gets a chance to create sample dataset JSON from `frontend/test/snapshot-creators/default.cy.snap.js`

**Comment by paulrosenzweig** [comment]
> My best guess is that it fails because linter is running first

If it's *only* the linter, you can add a comment to ignore that. However, the snapshot script will also import that file when it starts up. Maybe you could commit an empty file (or `{}`) that's overwritten with the real data? That will ensure the file can always be read and exported.

> Either that, or CI doesn't have permission to write files (which seems weird).

Snapshots rely on writing out SQL files for the backend, so that should be fine.

**Comment by nemanjaglumac** [comment]
> If it's _only_ the linter, you can add a comment to ignore that. However, the snapshot script will also import that file when it starts up. Maybe you could commit an empty file (or `{}`) that's overwritten with the real data? That will ensure the file can always be read and exported.

Before I saw your comment, I was searching for the ways to ignore this specific import and committed already :/. Let's see if that works first but maybe the idea with the empty file is cleaner. I just need to make sure it's committed first, but then I need to add it to the `.gitignore` afterwards.

**Comment by nemanjaglumac** [comment]
@paulrosenzweig I think this is ready for another inspection. I implemented your other suggestion and created another JS file that exports contents of this generated JSON file.

Everything seems to be working now. 🧙‍♂️

File: frontend/test/metabase/scenarios/admin/settings/localization.cy.spec.js [hunk 1]
```diff
@@ -1,4 +1,7 @@
-import { restore, withSampleDataset, signInAsAdmin } from "__support__/cypress";
+import { restore, signInAsAdmin } from "__support__/cypress";
+import { SAMPLE_DATASET } from "__support__/cypress_sample_dataset";
+
+const { ORDERS, ORDERS_ID } = SAMPLE_DATASET;
```
Comment (paulrosenzweig) [inline frontend/test/metabase/scenarios/admin/settings/localization.cy.spec.js:4]: 🥳

**Review by paulrosenzweig — APPROVED** [review]
Looks great!

File: frontend/test/metabase/scenarios/admin/settings/localization.cy.spec.js (hunk 1 above)
Comment (paulrosenzweig) [reply frontend/test/metabase/scenarios/admin/settings/localization.cy.spec.js:4]: Look at that! Access to the ids without nesting everything inside a callback! 🎉

Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/admin/settings/localization.cy.spec.js:4]: @paulrosenzweig you mean [THIS](https://github.com/metabase/metabase/pull/14004) 🤓

**Pull request merged**